Add how to use wrappers to README and refactor MetadataViews comments (#109)
onflow/flow-nft: README.md

| ----------- | ------------------------------------------ | ----------- | -------------------------------------------------------------------------------------------------------- |
| `Display`   | Return the basic representation of an NFT. | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L31-L66)  |
| `HTTPFile`  | A file available at an HTTP(S) URL.        | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L86-L98)  |
| `IPFSFile`  | A file stored in IPFS.                     | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L100-L137) |
| `Edition`   | Return information about one or more editions for an NFT. | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L139-L174) |
| `Editions`  | Wrapper for multiple edition views.        | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L176-L187)|

> | `Serial`    |                                            | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L200-L211)|

Reviewer: It looks like some of these are missing descriptions. Can you add some for the ones that are missing?

| `ExternalURL`|                                           | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L375-L385)|
| `NFTCollectionData` | Provides storage and retrieval information of an NFT | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L398-L455) |
| `NFTCollectionDisplay` | Returns the basic representation of an NFT's Collection.  | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L468-L507) |
| `Rarity`   |                                            | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L523-L550)|
| `Trait`    |                                            | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L563-L590)|
| `Traits`   |                                            | Implemented | [MetadataViews.cdc](https://github.com/onflow/flow-nft/blob/master/contracts/MetadataViews.cdc#L592-L605)|

## Allways prefer wrappers over single views

> When exposing a view that could have multiple ocurrences on a single NFT, such as `Edition`, `Royalty`, `Media` or `Trait` the wrapper view should allways be exposed, even if there is a single occurrence.

Reviewer: ```suggestion
When exposing a view that could have multiple occurrences on a single NFT, such as `Edition`, `Royalty`, `Media` or `Trait` the wrapper view should always be used, even if there is only a single occurrence. The wrapper view is always the plural version of the single view name and can be found below the main view definition in the `MetadataViews` contract.
```

> When resolving the view, the wrapper view should be the returned value, instead of returning the single view or just an array of several ocurrences of the view

Reviewer: ```suggestion
When resolving the view, the wrapper view should be the returned value, instead of returning the single view or just an array of several occurrences of the view.
```

### Example

#### Prefered

```cadence
pub fun resolveView(_ view: Type): AnyStruct? {
    switch view {
        case Type<MetadataViews.Editions>():
            let editionInfo = MetadataViews.Edition(name: "Example NFT Edition", number: self.id, max: nil)
            let editionList: [MetadataViews.Edition] = [editionInfo]
            return MetadataViews.Editions(
                editionList
            )
    }
}
```

#### To be avoided

```cadence

> pub fun resolveView(_ view: Type): AnyStruct? {

Reviewer: ```suggestion
// `resolveView` should always return the same type that was passed to it as an argument, so this is improper usage because it returns `Edition` instead of `Editions`.
pub fun resolveView(_ view: Type): AnyStruct? {
```
